### PR TITLE
Server shooting sync

### PR DESCRIPTION
**Pull request by dolanmiu**

### CODE REVIEW

File: src/main/client/bullet/bullet-group-component.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import * as Phaser from 'phaser';
+import { Inject, Singleton } from 'typescript-ioc';
+
+import { Player } from '../../shared/player/player-model';
+import { Subject } from 'rxjs';
+import { BulletFireOptions, Bullets } from './default-bullet';
+import { BulletStore } from '../../shared/bullet/bullet-store';
+
+@Singleton
+export class BulletGroupComponent {
```
Comment (barnabasbartha) [inline src/main/client/bullet/bullet-group-component.ts]: I think it should be ClientBulletComponent and ClientBulletManager

File: src/main/client/bullet/bullet-group-manager.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import { ClientNetworkComponent } from '../network/client-network-component';
+import { BulletStore } from '../../shared/bullet/bullet-store';
+
+@Singleton
+export class BulletGroupManager {
+   constructor(
+      @Inject private readonly component: BulletGroupComponent,
+      @Inject private readonly gameState: GameStateComponent,
+      @Inject private readonly network: ClientNetworkComponent,
+      @Inject private readonly bulletStore: BulletStore,
```
Comment (barnabasbartha) [inline src/main/client/bullet/bullet-group-manager.ts]: Dont inject components and others thing than the manager's component if you don't use them here.

File: src/main/server/network/server-network-model.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { NetworkMessage } from '../../shared/network/shared-network-model';
 import { IObject } from '../../shared/util/util-model';
 
-export interface ServerNetworkMessage<T extends IObject = IObject> extends NetworkMessage<T> {
+export interface ServerNetworkMessage<T = IObject> extends NetworkMessage<T> {
```
Comment (barnabasbartha) [inline src/main/server/network/server-network-model.ts:4]: What if it receivs an interface that extends T? Will we able to refer its properties?

**Review by barnabasbartha — CHANGES_REQUESTED** [review]
Resolve comments

File: src/main/client/scene/star-field-sprite.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      canvas.width = options.scene.game.scale.width * 2;
+      canvas.height = options.scene.game.scale.height * 2;
+
+      const texture = options.scene.textures.addCanvas('star-field', canvas);
+      super(options.scene, options.scene.game.scale.width / 2, options.scene.game.scale.height / 2, 'star-field');
+
+      for (let i = 0; i < 2000; i++) {
+         const radius = Math.random() * 2 + 0.2;
+         const x = Math.floor(Math.random() * canvas.width) + 1;
+         const y = Math.floor(Math.random() * canvas.height) + 1;
```
Comment (barnabasbartha) [inline src/main/client/scene/star-field-sprite.ts]: Is this a random between min and max? We could extract this to a util. I think I wrote something like this for previous projects:
`    static randomBetween(min: number, max: number): number {
        return Math.random() * (max - min) + min;
    }`

File: src/main/server/network/server-network-model.ts (hunk 3 above)
Comment (dolanmiu) [reply src/main/server/network/server-network-model.ts:4]: yeah, i wouldn't see why not

File: src/main/client/scene/star-field-sprite.ts (hunk 4 above)
Comment (dolanmiu) [reply src/main/client/scene/star-field-sprite.ts]: yeah good idea

**Review by barnabasbartha — APPROVED** [review]

**Pull request merged**